util/configmap_util.go

```diff
@@ -0,0 +1,44 @@
+package util
+
+import (
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+
+	errorsutil "github.com/argoproj/argo/util/errors"
+	"github.com/argoproj/argo/util/retry"
+)
+
+// GetConfigMaps retrieves a configmap value and memoizes the result
+func GetConfigMaps(clientSet kubernetes.Interface, namespace, name, key string) (string, error) {
+
+	configMapsIf := clientSet.CoreV1().ConfigMaps(namespace)
+	var configMap *apiv1.ConfigMap
+	var err error
```

Comment:
This err does not need to be scoped here. If we return an error within `ExponentialBackoff`, it will be retured on line 23. That line can just be

```go
err := wait.ExponentialBackoff(retry.DefaultRetry, func() (bool, error) {
```